### PR TITLE
Add Rule AddPrefix

### PR DESCRIPTION
**Pull request by Juliens**
Add the ability to add a prefix in the url when the backend is called.

example : 
```
[backends]

[backends.backend-test.servers.server-1]
url = "http:/backend-server"
weight = 1

[frontends]
[frontends.frontend-test]
    backend = "backend-test"
    entryPoints = ["http"]
[frontends.frontend-test.routes.route-host-test]
    rule = "PathPrefix:/test;PathPrefixStrip:/test;AddPrefix:/prefix"
```

when call http://traefik/test/url => forward http://backend-server/prefix/url

### CODE REVIEW

**Comment by emilevauge** [comment]
@Juliens thanks for this one :)
Could you add some documentation in `traefik.sample.toml` and in `docs/` ?

**Comment by emilevauge** [comment]
Ping @containous/traefik

**Comment by Russell-IO** [comment]
LGTM :+1:

File: docs/basics.md [hunk 1]
```diff
@@ -85,6 +85,7 @@ Frontends can be defined using the following rules:
 - `PathStrip`: Same as `Path` but strip the given prefix from the request URL's Path.
 - `PathPrefix`: PathPrefix adds a matcher for the URL path prefixes. This matches if the given template is a prefix of the full URL path.
 - `PathPrefixStrip`: Same as `PathPrefix` but strip the given prefix from the request URL's Path.
+- `AddPrefix` : Add prefix from the request URL's Path.
```
Comment (ralphtheninja) [inline docs/basics.md]: This section could really use some examples. (Generic comment, not related to this PR specifically)

**Pull request merged**